adapters/platform_linux.c

```diff
@@ -23,9 +23,43 @@ int platform_init(void)
     int result;
 #ifdef USE_OPENSSL
     result = tlsio_openssl_init();
+
+    char *host = getenv("HTTP_PROXY");
```

Comment:
is HTTP_PROXY a well-known environment variable for proxy setting on Linux/Windows? If not, we should not check this environment variable. If we do not clearly document this, user might have unexpected behavior if this variable is set by accident. Since the proxy data is pass from API level, there is no need to check this environment variable. And even worse, this could lead to inconsistence with the proxy used by websocket configured by the API.   
If we want to use system-wide proxy setting, this should be done at the top level (since websocket will need this setting anyway) and passed with set_http_proxy().

Reply:
for unix, it is. for windows, there is no such "generally available" environment variable.